src/elfpy/pricing_models.py

```diff
@@ -994,7 +1014,23 @@ def calc_out_given_in(
                 f"\n\tspot_price = {spot_price}\n\tk = {k}\n\twithout_fee_or_slippage = {without_fee_or_slippage}"
                 f"\n\twithout_fee = {without_fee}\n\twith_fee = {with_fee}\n\tfee = {fee}"
             )
-        assert fee >= 0, f"pricing_models.calc_out_given_in: ERROR: Fee should not be negative, not {fee}!"
+
+        # TODO(jalextowle): With some analysis, it seems possible to show that
+        # we skip straight from non-negative reals to the complex plane without
+        # hitting negative reals.
+        #
+        # Ensure that the outputs are all non-negative floats. We only need to
+        # check with_fee since without_fee_or_slippage will always be a positive
+        # float due to the constraints on the inputs, without_fee = with_fee + fee
+        # so it is a positive float if with_fee and fee are positive floats, and
+        # fee is a positive float due to the constraints on the inputs.
+        assert (
+            type(with_fee) == float
```

Comment:
I think it is better to use [isinstance](https://docs.python.org/3/library/functions.html#isinstance) instead. Also, might as well raise a TypeError instead of failed assertion.

Reply:
https://switowski.com/blog/type-vs-isinstance/